MonoGame.Framework/Plane.cs

```diff
@@ -30,8 +30,8 @@ public static float ClassifyPoint(ref Vector3 point, ref Plane plane)
         public static float PerpendicularDistance(ref Vector3 point, ref Plane plane)
         {
             // dist = (ax + by + cz + d) / sqrt(a*a + b*b + c*c)
-            return (float)Math.Abs((plane.Normal.X * point.X + plane.Normal.Y * point.Y + plane.Normal.Z * point.Z)
-                                    / Math.Sqrt(plane.Normal.X * plane.Normal.X + plane.Normal.Y * plane.Normal.Y + plane.Normal.Z * plane.Normal.Z));

```

Comment:
Precision lost!
The division was done with double precision because there was no cast to float in front of Math.Sqrt().